lib/validators/post_validator.rb

```diff
@@ -141,6 +142,26 @@ def unique_post_validator(post)
     end
   end
 
+  def force_edit_last_validator(post)
+    return if SiteSetting.max_consecutive_replies == 0 || post.id || post.acting_user.try(:staff?) || private_message?(post)
```

Comment:
Why `post.id`?

Reply:
Presence of `post.id` signals that the post already exists and this validator runs for an edited post, which should not be happen.

Reply:
```suggestion
    return if SiteSetting.max_consecutive_replies == 0 || post.id || post.acting_user&.staff? || private_message?(post)
```